presto-hive/src/main/java/io/prestosql/plugin/hive/HiveQlToPrestoTranslator.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.hive;
+
+import com.google.common.collect.Iterators;
+import com.google.common.collect.Lists;
+import com.google.common.collect.PeekingIterator;
+import io.prestosql.spi.PrestoException;
+
+import static io.prestosql.plugin.hive.HiveErrorCode.HIVE_INVALID_VIEW_DATA;
+import static java.lang.String.format;
+
+/**
+ * Translate statements in Hive QL to Presto SQL.
+ *
+ * Only translation of quoted literals is currently included.
+ */
+public final class HiveQlToPrestoTranslator
+{
+    // Translation methods consume data from the iterator
+    private final PeekingIterator<Character> input;
```

Comment:
You could make all these methods static, passing `input` and `output` through to each. I'm not sure what's cleaner style-wise, just a thought.

Reply:
I briefly considered it, but passing state as parameters felt like poor object-oriented design.

Alternatively, the state and non-static methods could be in a static nested class, which is what we have in some other places, like `ResolvedFunctionCallRewriter` and `ConnectorExpressionTranslator`. I started with that, but then it started to seem weird that nearly everything in the top-level class was part of the nested class.

Reply:
You know, passing state as parameters is "pure functional programming" which is more in fashion now than "object oriented programming" :)

Reply:
Well, yes, but this is also Java.

Reply:
Northeastern's CS program actually teaches a pure functional Java class before they do object oriented design.